designer/client/src/data/condition/updateConditions.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  hasConditionField,
+  hasConditionGroup,
+  type ConditionData,
+  type ConditionWrapper,
+  type FormDefinition
+} from '@defra/forms-model'
+
+import { getFields } from '~/src/data/component/fields.js'
+
+/**
+ * Update conditions to keep display text in sync with the component and section titles
+ */
+export function updateConditions(data: FormDefinition) {
+  const fields = getFields(data)
+
+  // Check if condition field display text is out of date
+  const checkCondition = (condition: ConditionData) => {
+    const fieldName = condition.field.name
+    const field = fields.find((input) => input.name === fieldName)
+
+    return field && condition.field.display !== field.label ? field : undefined
+  }
+
+  // Determine if any of the conditions need fixing
+  const hasConditionsChanged = data.conditions.some((wrapper) => {
+    return wrapper.value.conditions.some((condition) => {
+      if (hasConditionField(condition)) {
+        return checkCondition(condition)
+      } else if (hasConditionGroup(condition)) {
+        condition.conditions
+          .filter((condition) => hasConditionField(condition))
+          .forEach(checkCondition)
```

Comment:
Have we definitely got coverage for condition groups?

The result of this filter isn't returned anywhere

Reply:
Did you mean this?

```suggestion
        return condition.conditions
          .filter(hasConditionField)
          .some(checkCondition)
```

Reply:
Perhaps we move it to new `hasOutdatedCondition()` similar to `hasNestedCondition()`?

With some tests to make sure it correctly flags grouped-but-outdated conditions

https://github.com/DEFRA/forms-designer/blob/8570a50eb10ad66109040f811c36e5284c143d75/model/src/conditions/helpers.ts#L41-L43

Reply:
I think I'll keep it as-is for now. The function currently [depends](https://github.com/DEFRA/forms-designer/pull/594/files#diff-c4a25ac12bca23bbac8ac6afcf8c74955f56d66932edb22e1129fa4ff473b6a1R14) on the `FieldDef` class which currently lives in `designer/client/src/data/component/fields.ts`.

If an `hasOutdatedCondition` type function ever has any use outside this `fixupConditions` utiity function, let's move it then.

Reply:
No worries

>Have we definitely got coverage for condition groups?
>
>The result of this filter isn't returned anywhere

Can you add tests for groups though?

Just to confirm https://github.com/DEFRA/forms-designer/pull/594/commits/ee9f75717b3f9e5683af82927dae3da4e310999f works as expected

Reply:
Resolved with a73ab8aa5eb4296f92b5226e6b57497a1c48f149